.changeset/fluffy-actors-listen.md

```diff
@@ -0,0 +1,45 @@
+---
+'@leafygreen-ui/text-input': major
+---
+
+[LG-4145](https://jira.mongodb.org/browse/LG-4145)
+
+`aria-label` and `aria-labelledby` props are passed to the input.
+
+#### aria-labelledby
+`aria-labelledby` prop can only set the `aria-labelledby` attribute on the input if `label` prop is undefined. Otherwise, the generated label component id will be used for the `aria-labelledby` attribute
+
+| 👎 Does not use `aria-labelledby` prop | 👍 Does use custom `aria-labelledby` prop |
+| - | - |
+| `<TextInput label="Label" aria-labelledby="custom-label-id" />` | `<TextInput aria-labelledby="custom-label-id" />` |
```

Comment:
if both `aria-label` and `aria-labelledby` are set, what happens?

Reply:
I only covered these 2 cases for `aria-labelledby` because only `label` being defined will impact `aria-labelledby` being set

If both of those are set, that will affect `aria-label`, so I included an example in the below table showing examples for `aria-label`